.gemini/styleguide.md

```diff
@@ -0,0 +1,397 @@
+    the intent, the trade-offs, and the reasoning behind the implementation.
+
+*   _Comment sparingly_: Well-written code should be self-documenting where
+    possible. It's not necessary to add comments for code fragments that can
+    reasonably be assumed to be self-explanatory.
+
+*   _Use complete sentences_: Start comments with a capital letter and use
+    proper punctuation.
+
+### Overall nomenclature conventions
+
+This project follows some nomenclature rules and conventions in order to
+maintain a consistent interface that is easy to use. By using consistent naming,
+we can reduce cognitive load on human users and developers. Please try to use
+these terms when writing code.
+
+*   Use `state_vector` to describe a pure state.  **Do not** use `wavefunction`,
+    `wave_function`, or `state` for this object (`state` is too overloaded).
+
+*   If the object is an array or possibly a computational basis state
+    (given by an `int`), use `state_rep` or, if it is the initial state of
+    a system `initial_state`.
+
+*   A function argument (`state_vector`, `state_rep`, or `initial_state`)
+    should permit any of the possible representations of a state: A NumPy
+    array, a NumPy tensor, an integer representing a qubit-system's
+    computational basis state, a sequence of _n_ integers representing a
+    qudit's basis state, or a `cirq.ProductState`. The type annotation should
+    be `cirq.STATE_VECTOR_LIKE` and you should use `cirq.to_valid_state_vector`
+    to canonicalize as a NumPy array of amplitudes. If a function expects a
+    NumPy array of amplitudes, its type annotation should be `np.ndarray`.
+
+*   Use `density_matrix` to describe a mixed state that is passed in as a NumPy
+    matrix or NumPy tensor.  **Do not** used `mixed_state`, `density_operator`,
+    or `state`.
+
+## Python-specific guidance
+
+This section outlines the coding conventions for Python code in this project.
+
+### Python naming conventions
+
+*   _Variables_: Use lowercase with underscores (snake_case). Examples:
+    `qsim_op`, `qubit_to_index_dict`, `gate_kind`.
+
+*   _Constants_: Use uppercase with underscores. Examples: `MAX_VALUE`,
+    `DATABASE_NAME`.
+
+*   _Functions_: Use lowercase with underscores (snake_case). Examples:
+    `calculate_total()`, `process_data()`. Internal functions are prefixed with
+    an understore (`_`).
+
+*   _Classes_: Use CapWords (CamelCase). Examples: `UserManager`,
+    `PaymentProcessor`.
+
+*   _Modules_: Use lowercase with underscores (snake_case). Examples:
+    `user_utils`, `payment_gateway`.
+
+###  Docstrings and documentation
+
+This project uses [Google style doc
+strings](http://google.github.io/styleguide/pyguide.html#381-docstrings) with a
+Markdown flavor and support for LaTeX. Docstrings use tripe double quotes, and
+the first line should be a concise one-line summary of the function or object.
+
+Here is an example docstring:
+
+```python
+def some_method(a: int, b: str) -> float:
```

Comment:
nit - maybe `some_function` is better as method is usually associated with a class:
```suggestion
def some_function(a: int, b: str) -> float:
```